docs/cmdline-opts/haproxy-protocol.d

```diff
@@ -0,0 +1,11 @@
+Long: haproxy-protocol
```

Comment:
For convenience to command line users, do you think it would make sense to just call it `--haproxy` ?

Reply:
I would opt for --haproxy-protocol or --proxy-protocol, as the protocol could be used by other implementations too and is more descriptive.

Reply:
In my original PR I used --proxy-protocol and was recommended to include haproxy in the argument so as not to cause confusion with other proxy-related options.